lib/authz/permissions.go

```diff
@@ -1408,6 +1420,39 @@ func AuthorizeContextWithVerbs(ctx context.Context, log logrus.FieldLogger, auth
 	return authCtx, nil
 }
 
+// CheckAccessToKind will ensure that the user has access to the given verbs for the given kind.
+func (c *Context) CheckAccessToKind(quiet bool, kind string, verbs ...string) error {
+	ruleCtx := &services.Context{
+		User: c.User,
+	}
+
+	return c.CheckAccessToRule(quiet, ruleCtx, kind, verbs...)
+}
+
+// CheckAccessToResource will ensure that the user has access to the given verbs for the given resource.
+func (c *Context) CheckAccessToResource(quiet bool, resource types.Resource, verbs ...string) error {
+	ruleCtx := &services.Context{
+		User:     c.User,
+		Resource: resource,
+	}
+
+	return c.CheckAccessToRule(quiet, ruleCtx, resource.GetKind(), verbs...)
+}
+
+// CheckAccessToRule will ensure that the user has access to the given verbs for the given [services.Context] and kind.
+func (c *Context) CheckAccessToRule(quiet bool, ruleCtx *services.Context, kind string, verbs ...string) error {
+	if len(verbs) == 0 {
+		return trace.BadParameter("expected one or more verbs to be provided")
+	}
```

Comment:
What do you think about making the >= 1 verbs requirement explicit in the method signature? Compile errors >> runtime errors
```suggestion
func (c *Context) CheckAccessToRule(quiet bool, ruleCtx *services.Context, kind string, verb string, verbs ...string) error {
	verbs = append(verbs, verb)
```

Reply:
From the caller perspective I like the `(verb string, verbs ...string)` just because it's less typing and I don't have to make a slice if it's not needed. When can optimize the impl to something better than `verbs = append(verbs, verb)` if we're worried about extra allocations